File: daily/Linked List/1721-Swapping-nodes-in-a-linked-list.go
package daily

func swapNodes(head *ListNode, k int) *ListNode {
	left := head
	for i := 0; i < k-1; i++ {
		left = left.Next
	}

	fast := left
	slow := head
	for fast.Next != nil {
		slow = slow.Next
		fast = fast.Next
	}

	left.Val, slow.Val = slow.Val, left.Val
	return head
}

File: daily/Linked List/1721-Swapping-nodes-in-a-linked-list.py
class Solution:
    def swapNodes(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        left = head
        for _ in range(k - 1):
            left = left.next
        
        fast = left
        slow = head
        while fast.next:
            slow = slow.next
            fast = fast.next
        
        left.val, slow.val = slow.val, left.val
        return head